src/DagDerivedCompiler.cpp

```diff
@@ -311,36 +328,57 @@ struct CompileDagDerivedWorker
         {
             WriteArrayOfIndices(dependenciesArray_seg, combinedDependenciesBuffers[nodeIndex]);
             WriteArrayOfIndices(backlinksArray_seg, backlinksBuffers[nodeIndex]);
+        }
 
-            auto calculateCumulativePoints = [&](int nodeindex) -> uint32_t
+        {
+            TimingScope timing_scope(nullptr, &g_Stats.m_CumulativePointsTime);
+            Buffer<int32_t> all_scores;
+            BufferInitWithCapacity(&all_scores, heap, node_count);
+            for (int32_t nodeIndex = 0; nodeIndex < node_count; ++nodeIndex)
+                all_scores[nodeIndex] = -1;
+            for (int32_t nodeIndex = 0; nodeIndex < node_count; ++nodeIndex)
             {
-                std::function<void(int,Buffer<uint32_t>&)> add_backlinks_to_buffer = [&](int node_index, Buffer<uint32_t>& buffer)
+                //The algorithm we're going with here assigns a certain point score to each node, that will be used at runtime during scheduling.
+                //It's intended to (roughly) correspond to how much other work executing this node will unblock. In order to make the calculating
+                //of these scores not be too prohibitive, we are going with "amount of direct dependencies + the max of their scores".  This
+                //ensures that we will have as little as possible "long poles" in the build, and that the scheduler will work towards being able to
```

Comment:
"fewer scheduling bottlenecks" might be a more well known term

Reply:
I think long poles is a better description here (and a pretty common term, at least results in series of hits on google that are about this specific issue)

Reply:
TIL